### ISSUE TITLE
Proxy support

### ISSUE THREAD

**Issue by mrschyte** [htty/htty#126]
Please implement support for HTTP and SOCKS proxies. htty should pick up the HTTP_PROXY environment variable by default, and the user should be able to specify override this setting via the command line (e.g. --proxy http://localhost:8080 or --no-proxy).

**Comment by xfalcox** [comment]
Testing it here it does the standard ruby stuff:

- Picks HTTP_PROXY/HTPPS_PROXY env
- Parse this string, uses url and port, but discard user and password.

**Comment by mrschyte** [comment]
You're right, it does work with HTTP_PROXY. I had the lowercase version exported which works with curl, etc. but not with ruby applications.
